### PR TITLE
Aarch with cross

### PR DESCRIPTION
**Pull request by hmaarrfk**
- [x] Merge https://github.com/conda-forge/pyside2-feedstock/pull/132
- [x] Remove my channel

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/patches/do_not_store_build_time_compiler.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ static QString compilerFromCMake(const QString &defaultCompiler)
+ {
+-// Added !defined(Q_OS_DARWIN) due to PYSIDE-1032
+-#  if defined(CMAKE_CXX_COMPILER) && !defined(Q_OS_DARWIN)
+-    Q_UNUSED(defaultCompiler)
+-    return QString::fromLocal8Bit(CMAKE_CXX_COMPILER);
+-#  else
+-    return defaultCompiler;
+-#  endif
++    return qEnvironmentVariable("CXX", defaultCompiler);
```
Comment (hmaarrfk) [inline recipe/patches/do_not_store_build_time_compiler.patch:15]: @looooo I think this is what was broken for the cross compilation. Testing things locally, it seems to "build".

I'll try to upload a package to my channel. Would you be able to test it?

Comment (hmaarrfk) [reply recipe/patches/do_not_store_build_time_compiler.patch:15]: As a note, I'm really not sure what was broken in the "emulated" compilation. I expected it to take 6 hours and fail.

Comment (hmaarrfk) [reply recipe/patches/do_not_store_build_time_compiler.patch:15]: I'm building python 3.9 since it seem "innocent".

File: recipe/meta.yaml [hunk 2]
```diff
@@ -15,9 +15,12 @@ source:
     - patches/fix_genererate_pyi.patch  # [win]
     - patches/pyside2-tools-obsolete.patch
     - patches/cross_compile_pyi.patch  # [build_platform != target_platform]
+    - patches/do_not_store_build_time_compiler.patch
```
Comment (hmaarrfk) [inline recipe/meta.yaml:18]:
```suggestion
    # https://github.com/conda-forge/pyside2-feedstock/issues/130
    # https://bugreports.qt.io/browse/PYSIDE-2057
    - patches/do_not_store_build_time_compiler.patch
```

**Comment by hmaarrfk** [comment]
I think this is mostly "working" now. but I will need to fix ctest. It seems to be trying to run in the build environment.

File: recipe/patches/do_not_store_build_time_compiler.patch [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ }
+ #endif // Q_CC_CLANG, Q_CC_GNU
+ 
+diff -ur a_pyside-setup-opensource-src-5.15.6/sources/shiboken2/data/shiboken_helpers.cmake b_pyside-setup-opensource-src-5.15.6/sources/shiboken2/data/shiboken_helpers.cmake
+--- a_pyside-setup-opensource-src-5.15.6/sources/shiboken2/data/shiboken_helpers.cmake	2022-09-11 17:23:12.441907882 -0400
++++ b_pyside-setup-opensource-src-5.15.6/sources/shiboken2/data/shiboken_helpers.cmake	2022-09-11 17:30:26.057830906 -0400
+@@ -227,21 +227,7 @@
+   # We pick the first most detailed one.
+ 
+   execute_process(
```
Comment (hmaarrfk) [inline recipe/patches/do_not_store_build_time_compiler.patch:25]: Maybe I should take the code from Shiboken6:
```cmake
macro(get_python_extension_suffix)
    # When cross-building, we can't run the target python executable to find out the information,
    # so we rely on Python_SOABI being set by find_package(Python).
    # Python_SOABI is only set by CMake 3.17+
    # TODO: Lower this to CMake 3.16 if possible.
    if(SHIBOKEN_IS_CROSS_BUILD)
        if(NOT Python_SOABI)
            message(FATAL_ERROR "Python_SOABI variable is empty.")
        endif()
        set(PYTHON_EXTENSION_SUFFIX ".${Python_SOABI}")
    else()
        # See PYSIDE-1841 / https://bugs.python.org/issue39825 for distutils vs sysconfig
        execute_process(
          COMMAND ${PYTHON_EXECUTABLE} -c "if True:
             import sys
             if sys.version_info >= (3, 8, 2):
                 import sysconfig
                 suffix = sysconfig.get_config_var('EXT_SUFFIX')
             else:
                 from distutils import sysconfig
                 suffix = sysconfig.get_config_var('EXT_SUFFIX')
             pos = suffix.rfind('.')
             if pos > 0:
                 print(suffix[:pos])
             else:
                 print(f'Unable to determine PYTHON_EXTENSION_SUFFIX from EXT_SUFFIX: \"{suffix}\"',
                     file=sys.stderr)
             "
          OUTPUT_VARIABLE PYTHON_EXTENSION_SUFFIX
          OUTPUT_STRIP_TRAILING_WHITESPACE)
    endif()
    message(STATUS "PYTHON_EXTENSION_SUFFIX: " ${PYTHON_EXTENSION_SUFFIX})
endmacro()

```

**Comment by hmaarrfk** [comment]
<details> <summary>Getting closer</summary>


```
import: 'PySide2.QtCharts'
import: 'PySide2.QtConcurrent'
import: 'PySide2.QtCore'
import: 'PySide2.QtDataVisualization'
import: 'PySide2.QtGui'
import: 'PySide2.QtHelp'
import: 'PySide2.QtLocation'
import: 'PySide2.QtMultimedia'
import: 'PySide2.QtMultimediaWidgets'
import: 'PySide2.QtNetwork'
import: 'PySide2.QtOpenGL'
import: 'PySide2.QtPositioning'
import: 'PySide2.QtPrintSupport'
import: 'PySide2.QtQml'
import: 'PySide2.QtQuick'
import: 'PySide2.QtQuickWidgets'
import: 'PySide2.QtRemoteObjects'
import: 'PySide2.QtScript'
import: 'PySide2.QtScriptTools'
import: 'PySide2.QtScxml'
import: 'PySide2.QtSensors'
import: 'PySide2.QtSql'
import: 'PySide2.QtSvg'
import: 'PySide2.QtTest'
import: 'PySide2.QtTextToSpeech'
import: 'PySide2.QtUiTools'
import: 'PySide2.QtWebChannel'
import: 'PySide2.QtWebEngineCore'
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/pyside2_1662932735675/test_tmp/run_test.py", line 83, in <module>
    import PySide2.QtWebEngineCore
  File "/home/conda/feedstock_root/build_artifacts/pyside2_1662932735675/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.9/site-packages/shiboken2/files.dir/shibokensupport/feature.py", line 139, in _import
    return original_import(name, *args, **kwargs)
ImportError: libXdamage.so.1: cannot open shared object file: No such file or directory
Tests failed for pyside2-5.15.6-py39h8e67a9f_1.tar.bz2 - moving package to /home/conda/feedstock_root/build_artifacts/broken
WARNING:conda_build.build:Tests failed for pyside2-5.15.6-py39h8e67a9f_1.tar.bz2 - moving package to /home/conda/feedstock_root/build_artifacts/broken
TESTS FAILED: pyside2-5.15.6-py39h8e67a9f_1.tar.bz2
valid configs are {'linux_64_python3.9.____cpython', 'linux_aarch64_python3.9.____cpython'}
Using linux_aarch64_python3.9.____cpython configuration
Traceback (most recent call last):
  File "/home/mark/git/feedstocks/pyside2-feedstock/build-locally.py", line 105, in <module>
    main()
  File "/home/mark/git/feedstocks/pyside2-feedstock/build-locally.py", line 93, in main
    run_docker_build(ns)
  File "/home/mark/git/feedstocks/pyside2-feedstock/build-locally.py", line 29, in run_docker_build
    subprocess.check_call([script])
  File "/home/mark/mambaforge/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['.scripts/run_docker_build.sh']' returned non-zero exit status 1.

```
</details>

File: recipe/patches/do_not_store_build_time_compiler.patch (hunk 3 above)
Comment (hmaarrfk) [reply recipe/patches/do_not_store_build_time_compiler.patch]: I'm going to try to set the environment variable instead.

**Comment by hmaarrfk** [comment]
Here is a build that claims to work with "emulation" [log.zip](https://github.com/conda-forge/pyside2-feedstock/files/9543985/log.zip)

**Comment by hmaarrfk** [comment]
Builds available at https://anaconda.org/mark.harfouche/pyside2

**Comment by looooo** [comment]
great, thanks for your work @hmaarrfk :+1:

**Comment by looooo** [comment]
is there something missing @hmaarrfk ?

**Comment by hmaarrfk** [comment]
I guess i want to give the maintainers time to review: https://github.com/conda-forge/pyside2-feedstock/pull/132

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

aarch is expected to fail for at least an other 2 hours due to the need for new x86 builds

**Pull request merged**